concourse/scripts/pipecleaner.py

```diff
@@ -137,6 +164,18 @@ def check_pipeline(self, data):
                         for i in item['config']['outputs']:
                             output_resources.add(i['name'])
 
+                    if 'run' in item['config']:
+                        if item['config']['run']['path'] in ['sh', 'bash', 'dash', 'ksh']:
```

Comment:
Though we use it as `sh`, which is to say in ash_mode.

shellcheck doesn't have an `ash` mode.

~~~
$ shellcheck
No files specified.

Usage: shellcheck [OPTIONS...] FILES...
  -e CODE1,CODE2..  --exclude=CODE1,CODE2..  exclude types of warnings
  -f FORMAT         --format=FORMAT          output format
  -C[WHEN]          --color[=WHEN]           Use color (auto, always, never)
  -s SHELLNAME      --shell=SHELLNAME        Specify dialect (sh,bash,dash,ksh)
  -x                --external-sources       Allow 'source' outside of FILES.
  -V                --version                Print version information
~~~

I'll extract it to a `shellcheck_dialect` variable with a comment so it's obvious that's where the limitation comes from